app/bundles/PageBundle/Controller/PublicController.php

```diff
@@ -500,7 +500,11 @@ public function redirectAction($redirectId)
             }
 
             if (false !== strpos($url, $this->generateUrl('mautic_asset_download'))) {
-                $url .= '?ct='.$ct;
+                if (strpos($url, '&')) {
+                    $url .= '&ct=' . $ct;
+                } else {
+                    $url .= '?ct=' . $ct;
+                }
```

Comment:
Isn't this if opposite of what it should do? If it contains `&` then append the GET param with `&`? Also, check the warning in the docs for this function: https://www.php.net/manual/en/function.strpos.php

It's better to compare with `false` as the `0` value can cause unwanted behavior.